### ISSUE TITLE
Zones act randomly

### ISSUE THREAD

**Issue by manuxio** [MatteoManzoni/RehauNeasmart2.0_Gateway#18]
Zone presets act very randomly and keep activating by themselves.
Unfortunately this makes it impossible to use the addon.
Can you please test it?
I'm using a Waveshare adapter
The same random effect can be seen as soon as the pymodbus server gets a connection
Some tests with pyModbusTCP, on the other hand, seems to be fine

I think the same odd behavious was previously reported by https://github.com/MatteoManzoni/RehauNeasmart2.0_Gateway/issues/14

**Comment by bardaxx** [comment]
hey @manuxio could you please provide steps to reproduce the issue?

**Comment by manuxio** [comment]
Hello,
unfortunately I had to stop using this local integration in favour of the cloud based one because of the bug described in this issue.
If you feel like actively taking care of the project I might commit to mount the device back and provide more feedbacks.
If you don't have enough time, I can't take the chance.
Let me know!

**Comment by zbuh** [comment]
@manuxio There is another integration?

**Comment by manuxio** [comment]
This a fork with some changes I made https://github.com/manuxio/rehau-nea-smart-2.0-ha
The original repo is https://github.com/th3r3alandr3/rehau-nea-smart-2.0-ha

I believe the code was developed by reverse engineering the android apk.
It basically authenticates and then uses the remote mqtt to which ALL rehau installations in the world are connected!

Far from desirable (in terms of privacy and so on) but definitely works fine.

**Comment by zbuh** [comment]
Thanks for the info. I did not knew that instegration.
Tried to install it but it seems broken, maybe due to HACS changes?!?

I really prefer the local integration, but without fixing this issue with states save/restore, the cloud solution could be a solution.

The error in hacs:

```
This error originated from a custom integration.

Logger: custom_components.hacs
Source: custom_components/hacs/base.py:709
integration: HACS (documentation, issues)
First occurred: 08:09:51 (2 occurrences)
Last logged: 08:14:59

Download failed - Got status code 404 when trying to download https://github.com/manuxio/rehau-nea-smart-2.0-ha/releases/download/main/rehau_nea_smart_2.zip
Traceback (most recent call last):
  File "/config/custom_components/hacs/base.py", line 709, in async_download_file
    raise HacsException(
custom_components.hacs.exceptions.HacsException: Got status code 404 when trying to download https://github.com/manuxio/rehau-nea-smart-2.0-ha/releases/download/main/rehau_nea_smart_2.zip
```

**Comment by bardaxx** [comment]
> Hello, unfortunately I had to stop using this local integration in favour of the cloud based one because of the bug described in this issue. If you feel like actively taking care of the project I might commit to mount the device back and provide more feedbacks. If you don't have enough time, I can't take the chance. Let me know!

Hi Manuxio, thanks for the feedback, I understand your point.
I’m currently working on improving the repo and am in contact with Matteo. I’ve installed the Waveshare and connected it to the HA addon—everything is working fine, but I’m still figuring out some details. Since I’m running Home Assistant as a core on a Docker container, the addon doesn’t work for me, so I’ve started a [new codebase](https://github.com/giacomoalonzi/RehauNeasmart2.0_Gateway).

My goal is to eventually merge my code into the original with some fixes and improvements. Any help or suggestions are welcome. When you have time, please explain the bug in more detail, and I’ll see if there’s a way to find and fix it.

**Comment by manuxio** [comment]
> My goal is to eventually merge my code into the original with some fixes and improvements. Any help or suggestions are welcome. When you have time, please explain the bug in more detail, and I’ll see if there’s a way to find and fix it.

Hello,
as I said, I am willing to help if you commit to your project. 
Having the chiller/boiler acting randomly is not easy to be accepted by the family ... 

I totally agree that local integration is far more desirable.

If you plan to do so I will be available to for some time to provide test/help/whatever

M.

**Comment by bardaxx** [comment]
> > My goal is to eventually merge my code into the original with some fixes and improvements. Any help or suggestions are welcome. When you have time, please explain the bug in more detail, and I’ll see if there’s a way to find and fix it.
> 
> Hello, as I said, I am willing to help if you commit to your project. Having the chiller/boiler acting randomly is not easy to be accepted by the family ...
> 
> I totally agree that local integration is far more desirable.
> 
> If you plan to do so I will be available to for some time to provide test/help/whatever
> 
> M.

Absolutely! This is my home heating system for life in this house, so it has to work flawlessly! 😂 I’m currently trying to figure out why I can’t open and close the circuits via Home Assistant. I am receiving the temperature and humidity sensor data, but I’m unable to control individual rooms.

What's the best way for you to keep in touch? Email? I regularly use discord as chat client. Let me know

**Comment by Gsurg** [comment]
Hello guys, have nea smart 2 in my new home and as you disappointed with the smart side of things so trying to integrate with my solar so I can heat the house when we have excess power etc.

I'll be trying this over the next week's and months of you need some more assistance.

**Comment by ashebbeare** [comment]
Hi @MatteoManzoni - just wanted to flag this discussion to you - https://github.com/manuxio/rehau-nea-smart-2-home-assistant/issues/23

Seems like @tiagoroldao might have figured out the unit-0 broadcast thing - since this was all based on your project I wanted to make sure you were aware. I am still waiting for my waveshare device to test myself
